stellargraph/core/graph.py

```diff
@@ -279,6 +274,25 @@ def __init__(
             weight_column=edge_weight_column,
         )
 
+        edge_node_set = set(self._edges.sources)
```

Comment:
I think doing the switch from Pandas/NumPy world to Python here is going to be quite expensive, especially because it happens unconditionally. What does it do to `test_benchmark_creation` and `test_allocation_benchmark_creation`? I think you can have an automatic analysis by first running on develop like

```
git checkout origin/develop
pytest -k benchmark_creation --benchmark-autosave
```

and then running on this branch
```
git checkout enhancement/894-no-node-error-handling
pytest -k benchmark_creation --benchmark-autosave --benchmark-compare
```

(There's [a `pytest-benchmark compare ...` command](https://pytest-benchmark.readthedocs.io/en/latest/comparing.html#comparing-outside-of-pytest) for retroactive analysis too.)

Reply:
**Results fo this branch**

![image](https://user-images.githubusercontent.com/23410124/76587732-fd45b780-6538-11ea-8d6e-4508b74370c0.png)

Reply:
**Results for develop**

![image](https://user-images.githubusercontent.com/23410124/76587809-354cfa80-6539-11ea-81fe-3dd850910cb4.png)

Reply:
looks like this does slow things down - I'll work on a pandas/numpy approach

Reply:
That's not much of a slow down. What about the memory benchmarks?